src/main/client_context.cpp

```diff
@@ -590,6 +590,16 @@ PendingExecutionResult ClientContext::ExecuteTaskInternal(ClientContextLock &loc
 			active_query->progress_bar->Update(is_finished);
 			query_progress = active_query->progress_bar->GetDetailedQueryProgress();
 		}
+
+		if (progress_callback) {
+			duckdb_query_progress_type progress;
```

Comment:
This sounds like logic that should be performed on the C side, in the `progress_callback`, not here